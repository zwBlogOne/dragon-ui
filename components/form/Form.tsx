import React, { Component } from 'react';
import classnames from 'classnames';
import PropsType from './PropsType';

class Form extends Component<PropsType, any> {
  static defaultProps = {
    prefixCls: 'ui-form',
    type: 'horizontal',
    className: null,
  };
  static Item;
  render() {
    const {
      type, className, children, style, prefixCls,
    } = this.props;

    const cls = classnames({
      [prefixCls!]: true,
      [`${prefixCls}-${type}`]: 'type' in this.props,
      [className!]: !!className,
    });

    return (
      <form className={cls} style={style}>
        {children}
      </form>
    );
  }
}

export default Form;
