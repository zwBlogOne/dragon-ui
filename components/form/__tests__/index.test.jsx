import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Form from '../index';
import Input from '../../input';
import Button from '../../button';

describe('Form', () => {
  it('renders normal Form correctly', () => {
    const wrapper = render(
      <Form>
        <Form.Item label="账号">
          <Input placeholder="请输入..." />
        </Form.Item>
        <Form.Item label="密码">
          <Input placeholder="请输入..." />
        </Form.Item>
        <Form.Item>
          <Button theme="success">登录</Button>
        </Form.Item>
      </Form>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders inline Form with label correctly', () => {
    const wrapper = render(
      <Form type="inline">
        <Form.Item
          className="col-sm-4"
          label="类型"
        >
          <Input placeholder="请输入..." />
        </Form.Item>
        <Form.Item
          className="col-sm-4"
          label="来源"
        >
          <Input placeholder="请输入..." />
        </Form.Item>
        <Form.Item
          className="col-sm-4"
          label=""
        >
          <Button theme="success">查询</Button>
        </Form.Item>
      </Form>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders inline Form with required fields and help correctly', () => {
    const wrapper = render(
      <Form type="inline">
        <Form.Item
          className="col-sm-4"
          label="类型"
          help="什么是类型"
        >
          <Input placeholder="请输入..." />
        </Form.Item>
        <Form.Item
          className="col-sm-4"
          label="来源"
          required
        >
          <Input placeholder="请输入..." />
        </Form.Item>
        <Form.Item
          className="col-sm-4"
          label=""
        >
          <Button theme="success">查询</Button>
        </Form.Item>
      </Form>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
