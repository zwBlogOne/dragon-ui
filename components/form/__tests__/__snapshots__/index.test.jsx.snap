// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form renders inline Form with label correctly 1`] = `
<form
  class="ui-form ui-form-inline"
>
  <div
    class="ui-form-item col-sm-4"
  >
    <label
      class=""
    >
      类型
    </label>
    <div
      class="ui-form-item-control theme-default"
    >
      <span>
        <input
          class="ui-input"
          placeholder="请输入..."
          type="text"
        />
      </span>
    </div>
  </div>
  <div
    class="ui-form-item col-sm-4"
  >
    <label
      class=""
    >
      来源
    </label>
    <div
      class="ui-form-item-control theme-default"
    >
      <span>
        <input
          class="ui-input"
          placeholder="请输入..."
          type="text"
        />
      </span>
    </div>
  </div>
  <div
    class="ui-form-item col-sm-4"
  >
    <label
      class=""
    />
    <div
      class="ui-form-item-control theme-default"
    >
      <button
        class="ui-button theme-success"
        type="button"
      >
        查询
      </button>
    </div>
  </div>
</form>
`;

exports[`Form renders inline Form with required fields and help correctly 1`] = `
<form
  class="ui-form ui-form-inline"
>
  <div
    class="ui-form-item col-sm-4"
  >
    <label
      class=""
    >
      类型
    </label>
    <div
      class="ui-form-item-control ui-form-item-control-help theme-default"
    >
      <span>
        <input
          class="ui-input"
          placeholder="请输入..."
          type="text"
        />
      </span>
      <div
        class="ui-form-explain"
      >
        什么是类型
      </div>
    </div>
  </div>
  <div
    class="ui-form-item col-sm-4"
  >
    <label
      class=""
    >
      <span
        class="ui-form-item-required"
      >
        <i
          class="ui-icon ui-icon-required theme-default"
        />
      </span>
      来源
    </label>
    <div
      class="ui-form-item-control theme-default"
    >
      <span>
        <input
          class="ui-input"
          placeholder="请输入..."
          type="text"
        />
      </span>
    </div>
  </div>
  <div
    class="ui-form-item col-sm-4"
  >
    <label
      class=""
    />
    <div
      class="ui-form-item-control theme-default"
    >
      <button
        class="ui-button theme-success"
        type="button"
      >
        查询
      </button>
    </div>
  </div>
</form>
`;

exports[`Form renders normal Form correctly 1`] = `
<form
  class="ui-form ui-form-horizontal"
>
  <div
    class="ui-form-item"
  >
    <label
      class=""
    >
      账号
    </label>
    <div
      class="ui-form-item-control theme-default"
    >
      <span>
        <input
          class="ui-input"
          placeholder="请输入..."
          type="text"
        />
      </span>
    </div>
  </div>
  <div
    class="ui-form-item"
  >
    <label
      class=""
    >
      密码
    </label>
    <div
      class="ui-form-item-control theme-default"
    >
      <span>
        <input
          class="ui-input"
          placeholder="请输入..."
          type="text"
        />
      </span>
    </div>
  </div>
  <div
    class="ui-form-item"
  >
    <div
      class="ui-form-item-control theme-default"
    >
      <button
        class="ui-button theme-success"
        type="button"
      >
        登录
      </button>
    </div>
  </div>
</form>
`;
